src/main.js

```diff
@@ -1,57 +1,50 @@
-    const timeoutMilliseconds = 10 * 60 * 1000;
-
-    // determine the comment line number
-    var lineNumber;
-    if (isEmptyString(commentStartLine) || commentStartLine === "-1") {
-        console.log("Processing a single-line comment.");
-        lineNumber =  commentLine;
-    } else {
-        console.log("Processing a multi-line comment.");
-        lineNumber = commentStartLine;
-    }
 
-    // run CodeMaker generation
+    const owner = github.context.payload.repository.owner.login;
+    const repository = github.context.payload.repository.name;
+    const branch = github.context.payload.pull_request["head"]["ref"];
+
+    const pullRequestNumber = github.context.payload.pull_request.number;
+    const reviewId = github.context.payload["review"]["id"];
+
+    console.log(`Processing code review ${reviewId} on pull request ${pullRequestNumber}.`);
+
+    const comments = await getReviewComments(accessToken, owner, repository, pullRequestNumber, reviewId);
+    orderComments(comments);
+
     const client = new Client(() => apiKey);
-    const language = langFromFileExtension(commentFilePath);
-    console.log(`Running code generation for file ${commentFilePath}. Language detected: ${language}. Line number: ${lineNumber}. Prompt: ${commentBody}`);
 
-    console.log("Read original source code");
-    const source = await readFile(commentFilePath, 'utf8');
+    const modified = new Set();
+    for (let i = 0; i < comments.length; i++) {
+        const comment = comments[i];
+        const path = comment.path;
+        const line = lineNumber(comment);
+        const language = langFromFileExtension(path);
+        console.log(`Running code generation for file ${path}. Language detected: ${language}. Line number: ${line}. Prompt: ${comment.body}`);
+
+        const source = await readFile(path, 'utf8');
+        const request = createProcessRequest(language, source, line, comment.body);
+        const generatedSource = await processTask(client, request);
 
-    const request = createProcessRequest(language, source, lineNumber, commentBody);
-    
-    console.log(`Run generation. request: ${stringifyObjectAndReplaceValues(request,new Map([
-        ["source", "NOT_PRINTING_FULL_SOURCE_CODE"]
-    ]))}`);
-    const generatedSource = await processTask(client, request, pollingIntervalMilliseconds, timeoutMilliseconds);
+        console.log("Write generated code back to file");
+        await writeFile(path, generatedSource);
 
-    console.log("Write generated code back to file");
-    await writeFile(commentFilePath, generatedSource);
+        modified.add(path);
+    }
 
     console.log("Commit and push");
-    await commitAndPushSingleFile(accessToken, owner, repository, branch, commentFilePath,
-        "Updating PR according to comment: " + commentBody);
+    const paths = [...modified]
+    await commitAndPushSingleFile(accessToken, owner, repository, branch, paths, "Updated PR base on comments");
```

Comment:
could add more info in the commit msg, otherwise the commit history will be all the same

e.g. add comment author, comment date etc.